### PR TITLE
Refactoring.

### PR DESCRIPTION
**Pull request by FeniksFire**
Splitting CBattleCallback into smaller files and moving it with some others files to the "battle" subdirectory.

### CODE REVIEW

File: lib/battle/CBattleInfoEssentials.cpp [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+ *
+ */
+#include "StdInc.h"
+#include "CBattleInfoEssentials.h"
+#include "CStack.h"
+#include "BattleInfo.h"
+#include "NetPacks.h"
+#include "mapObjects/CGTownInstance.h"
+
+#define RETURN_IF_NOT_BATTLE(X) if(!duringBattle()) {logGlobal->errorStream() << __FUNCTION__ << " called when no battle!"; return X; }
```
Comment (FeniksFire) [inline lib/battle/CBattleInfoEssentials.cpp]: I completely forgot about this line, it is in 3 files, I will improve it.

File: lib/battle/CBattleInfoEssentials.cpp [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+	RETURN_IF_NOT_BATTLE(-1);
+	return getBattle()->tacticsSide;
+}
+
+const CGHeroInstance * CBattleInfoEssentials::battleGetFightingHero(ui8 side) const
+{
+	RETURN_IF_NOT_BATTLE(nullptr);
+	if(side > 1)
+	{
+		logGlobal->errorStream() << "FIXME: " <<  __FUNCTION__ << " wrong argument!";
```
Comment (ArseniyShestakov) [inline lib/battle/CBattleInfoEssentials.cpp]: Rewrite all logging to new format:
`logGlobal->error("Something %s something", __FUNCTION__);`

File: lib/battle/CBattleInfoEssentials.cpp [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+	if(!battleDoWeKnowAbout(side))
+	{
+		logGlobal->errorStream() << "FIXME: " <<  __FUNCTION__ << " access check ";
+		return nullptr;
+	}
+
+	return getBattle()->sides[side].armyObject;
+}
+
+InfoAboutHero CBattleInfoEssentials::battleGetHeroInfo( ui8 side ) const
```
Comment (ArseniyShestakov) [inline lib/battle/CBattleInfoEssentials.cpp]: Formatting of "( ".

File: lib/battle/CBattleInfoEssentials.cpp [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+	return getBattle()->sides.at(playerToSide(battleGetOwner(stack))).hero;
+}
+
+bool CBattleInfoEssentials::battleMatchOwner(const CStack * attacker, const CStack * defender, const boost::logic::tribool positivness /* = false*/) const
+{
+	RETURN_IF_NOT_BATTLE(false);
+	if(boost::logic::indeterminate(positivness))
+		return true;
+	else if(defender->owner != battleGetOwner(defender))
+		return true;//mind controlled unit is attackable for both sides
```
Comment (ArseniyShestakov) [inline lib/battle/CBattleInfoEssentials.cpp]: Add space before "//" when it's come on same line with code.

File: lib/battle/CBattleInfoEssentials.h [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+	ALL_KNOWING = -1,
+	LEFT_SIDE,
+	RIGHT_SIDE
+};
+}
+
+namespace BattleSide
+{
+enum {ATTACKER = 0, DEFENDER = 1};
+}
```
Comment (ArseniyShestakov) [inline lib/battle/CBattleInfoEssentials.h]: Indent namespaces with tabs. Some probably could be replaced with strongly typed enum.

**Comment by ArseniyShestakov** [comment]
BattleHex, BattleInfo, CBattleInfoCallback could still have few more fixes on formatting.
There is both spaces between brackets and some weird spaces in indent.

Other than that everything more or less fine.

**Comment by FeniksFire** [comment]
@alexvins I'm not sure if it is good now, since I did it manually. Check it if you can :).

**Comment by alexvins** [comment]
It`s ok if it still has some issues.

**Pull request merged**